semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,477 @@
+        )
+
+    def geertsma_ts_simple(self, vintage_pairs):
+        """
+        Calculates TS without using velocity. Fast.
+        Velocity is only used to get the surface on the velocity grid.
+
+        :param vintage_pairs:
+        """
+        return self._geertsma_ts_custom(
+            vintage_pairs, self.subsidence.evalGeertsma, "TS_SIMPLE"
+        )
+
+    def _geertsma_ts_custom(self, vintage_pairs, subsidence_func, method_name):
+        """
+        Calculates TS without using velocity. Fast.
+
+        :param vintage_pairs:
+        :param subsidence_func: specify subsidence method to be used
+        :param method_name: string represeting the subsudence func name
+        """
+
+        if len(vintage_pairs) < 1:
+            return 0, []
+
+        vintages = self._vintages_name_date(vintage_pairs)
+        surface = self._surface
+        points_to_calculate = self._get_non_nan_points()
+        surf_displacement = np.zeros((len(vintages.name), len(surface)))
+        ts_surfaces = []
+        for idx_vtg, (name_vtg, date_vtg) in enumerate(zip(vintages.name, vintages.date)):
+            try:
+                logging.info(
+                    "{:%x %X} {}: Calculating vintage {:%Y.%m.%d}".format(
+                        dt.now(), method_name, date_vtg
+                    )
+                )
+            except TypeError:
+                logging.exception("Wrong date vintage format!")
+                raise ValueError
+
+            self.add_survey(name_vtg, date_vtg)
+
+            for point in points_to_calculate:
+                r1 = (surface.x[point], surface.y[point], 0)
+                r2 = (
+                    surface.x[point],
+                    surface.y[point],
+                    surface.z[point] - self._seabed,
+                )
+                # subsidence and displacement have opposite sign
+                # should have minus on dz1 and dz2 here, more efficient
+                # when calculating ts
+                dz1 = subsidence_func(
+                    base_survey=name_vtg,
+                    monitor_survey=None,
+                    pos=r1,
+                    youngs_modulus=self._youngs_modulus,
+                    poisson_ratio=self._poisson_ratio,
+                    seabed=self._seabed,
+                )
+                dz2 = subsidence_func(
+                    base_survey=name_vtg,
+                    monitor_survey=None,
+                    pos=r2,
+                    youngs_modulus=self._youngs_modulus,
+                    poisson_ratio=self._poisson_ratio,
+                    seabed=self._seabed,
+                )
+                surf_displacement[idx_vtg, point] = dz2 - dz1
+
+        for pair in vintage_pairs:
+            self._report(method_name, pair[0], pair[1], len(points_to_calculate))
+
+            base_index = vintages.date.index(pair[0])
+            monitor_index = vintages.date.index(pair[1])
```

Comment:
Out of curiosity, the content of the vintage pairs at this point is that the first item is the base, while the second is the monitor. However in `geertsma_ts` it is the opposite?

```
self.add_survey("base", pair[1])
self.add_survey("monitor", pair[0])
```

I mean, one would expect such a mistake, if it really is, to have been caught a long time ago.. But it is without a doubt weird that we are not handling the vintages consistently.

Reply:
Exactly my point as well already added to questions!

Reply:
Appending an answer from Thomas:

Reversed computation for the dates fo base and monitor might be connected to the fact that Snorre uses reversed convention (and Snorre where first user of this workflow).

Reply:
So we are definitively changing this one then. I'm a bit confused as to why there's a `convention` parameter, and still this exists. But perhaps the change was added and the parameter followed afterwards without reverting how the function runs..

Never the less, let's keep consistent! :)

Reply:
Still waiting for the reply from Thomas to check whether `convention` alone suffices for the reversed case.

Reply:
I've gotten the reply, which states that the implementation should be correct and as such it's indeed reversed in `geertsma_ts` job.

Reply:
Alright.. Could be good with a comment, but I don't have anything better than
# Reversed because reasons

:P

Reply:
I don't really know what else to do, but to leave it as is.

Reply:
https://github.com/equinor/semeio/issues/189